DeepSkyStacker/TIFFUtil.cpp

```diff
@@ -12,29 +12,29 @@
 static const TIFFFieldInfo DSStiffFieldInfo[NRCUSTOMTIFFTAGS] =
 {
     { TIFFTAG_DSS_NRFRAMES,	1, 1, TIFF_LONG,	FIELD_CUSTOM,
-      false,	false,	"DSSNumberOfFrames" },
+      false,	false,	const_cast<char *>("DSSNumberOfFrames") },
```

Comment:
These `const_cast<char *>` are probably the thing I'm the most hesitant about: they don't change the existing behaviour, but it seems to me that passing a string literal to a struct that expects a non-const `char *` field is quite dangerous. If for any reason a piece of code attempts to modify the underlying string or `free()` this `char *`, I believe there is undefined behaviour.

But again, since this typing change does not modify the actual behaviour and since it seemed to work before, I guess it is "safe"?

Reply:
Hi @mtoeltsch, I fully agree that it's dangerous. (Just in case I didn't make it clear, that this was already committed and I did not change the behaviour in any way, I just satisfied the typechecker.)

> Also note how the definition of this struct is marked as OBSOLETE, and "Don't use this stuff in your applications".
Maybe a good time to get rid of it?

Indeed I saw that, I didn't want to make any actual business logic change when doing this PR because I wasn't familiar with the codebase and didn't want to break anything, but if you say it could be a good time to get rid of it I'll have a second look and try and push a PR for it.

Reply:
Hi Erwan, sorry if I was unfriendly, it wasn't meant like that. 
I know that this dangerous code was committed long ago, I just wanted to say that it might be a good time to check, if we can improve it. If we (you ;-) ) want to do that, we could try to upgrade the libtiff to the latest version. BUT: I think we have more important things to do.